sdk/src/types/block/semantic.rs

```diff
@@ -237,39 +230,83 @@ pub fn semantic_validation(
 
     // Validation of inputs.
     for ((output_id, consumed_output), unlock) in inputs.iter().zip(unlocks.iter()) {
-        let (conflict, amount, consumed_native_tokens, unlock_conditions) = match consumed_output {
+        let (conflict, amount, mana, consumed_native_tokens, unlock_conditions) = match consumed_output {
             Output::Basic(output) => (
                 output.unlock(output_id, unlock, inputs, &mut context),
                 output.amount(),
+                output.mana(),
                 Some(output.native_tokens()),
                 output.unlock_conditions(),
             ),
             Output::Account(output) => (
                 output.unlock(output_id, unlock, inputs, &mut context),
                 output.amount(),
+                output.mana(),
                 Some(output.native_tokens()),
                 output.unlock_conditions(),
             ),
             Output::Foundry(output) => (
                 output.unlock(output_id, unlock, inputs, &mut context),
                 output.amount(),
+                0,
                 Some(output.native_tokens()),
                 output.unlock_conditions(),
             ),
             Output::Nft(output) => (
                 output.unlock(output_id, unlock, inputs, &mut context),
                 output.amount(),
+                output.mana(),
                 Some(output.native_tokens()),
                 output.unlock_conditions(),
             ),
             Output::Delegation(output) => (
                 output.unlock(output_id, unlock, inputs, &mut context),
                 output.amount(),
+                0,
                 None,
                 output.unlock_conditions(),
             ),
         };
 
+        if let Output::Account(consumed) = consumed_output {
```

Comment:
Have a look at the `StateTransitionVerifier` impls for the chain outputs. They have a `destruction` function that is already being called by semantic validation. You could add this check there to avoid checking the outputs yourself.

Reply:
Ah, okay